### PR TITLE
[accordion] Add the accordion component

### PR DESCRIPTION
**Pull request by GianlucaGuarini**
I was able to get the 100% coverage on this component ;)

### CODE REVIEW

File: src/components/accordion/accordion.spec.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                isPassive: true,
+                isOpen: false,
+            },
+            scopedSlots: {
+                head: '<h1 class="label">Head</h1>',
+                body: '<p>Body</p>',
+            },
+            listeners: {
+                changed(isOpen) {
+                    expect(wrapper.vm.state.isOpen).to.be.equal(isOpen);
```
Comment (nirazul) [inline src/components/accordion/accordion.spec.js]: Can you change this to `true` or `false` according to the test you are running? The test should be as naive as possible, else we have a dependency on a variable defined in the test.

File: src/components/accordion/accordion.vue [hunk 2]
```diff
@@ -0,0 +1,121 @@
+<template>
+    <div class="ui-accordion" role="heading" :class="rootClasses">
```
Comment (nirazul) [inline src/components/accordion/accordion.vue]: Is the `role="heading"` really on this element? The body slot is also within it.

File: src/components/accordion/accordion.vue [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                role="presentation"
+                ref="body"
+                @transitionend="onTransitionEnd">
+            <slot name="body" :is-open="state.isOpen" :is-visible="isVisible"/>
+        </div>
+    </div>
+</template>
+
+<script>
+    import forceReflow from 'bianco.force-reflow';
```
Comment (nirazul) [inline src/components/accordion/accordion.vue]: Maybe we should add bianco as a whole. Else we need to add every single module in the webpack config.

File: src/components/accordion/accordion.vue [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                    isAnimating: false,
+                    isOpen: this.isOpen,
+                },
+            };
+        },
+        computed: {
+            isVisible() {
+                return this.isClosing || this.hasFinishedOpeningAnimation;
+            },
+            hasFinishedOpeningAnimation() {
```
Comment (nirazul) [inline src/components/accordion/accordion.vue]: maybe `hasOpened` is enough?

File: src/components/accordion/accordion.vue [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                const isTargetProperty = event.propertyName === TRANSITION_END_KEY_PROP;
+
+                if (!isTargetElement || !isTargetProperty) {
+                    return;
+                }
+
+                this.finishAnimation();
+                this.$emit('changed', this.state.isOpen);
+            },
+            finishAnimation() {
```
Comment (nirazul) [inline src/components/accordion/accordion.vue]: Can you move this function up to `startAnimation`? Then we have methods and listeners packed together.

**Comment by faebeee** [comment]
LGTM once again =)

File: src/components/accordion/accordion.spec.js (hunk 1 above)
Comment (GianlucaGuarini) [reply src/components/accordion/accordion.spec.js]: I have made this test a bit more explicit

File: src/components/accordion/accordion.vue (hunk 2 above)
Comment (GianlucaGuarini) [reply src/components/accordion/accordion.vue]: i have removed the accessibility attributes for now

File: src/components/accordion/accordion.vue (hunk 4 above)
Comment (GianlucaGuarini) [reply src/components/accordion/accordion.vue]: I have removed the computed and replaced them with variables

File: src/components/accordion/accordion.vue (hunk 5 above)
Comment (GianlucaGuarini) [reply src/components/accordion/accordion.vue]: I have removed the computed and replaced them with variables

File: src/components/accordion/accordion.vue (hunk 3 above)
Comment (GianlucaGuarini) [reply src/components/accordion/accordion.vue]: I have update the code using only one bianco dependency

**Pull request merged**